pkg/render/csi.go

```diff
@@ -290,6 +300,77 @@ func (c *csiComponent) csiDaemonset() *appsv1.DaemonSet {
 	}
 }
 
+func (c *csiComponent) serviceAccount() *corev1.ServiceAccount {
+
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      CSIDaemonSetName,
+			Namespace: CSIDaemonSetNamespace,
+		},
+	}
+}
+
+// podSecurityPolicy sets up a PodSecurityPolicy for CSI Driver to allow usage of privileged
+// securityContext and hostPath volume.
+func (c *csiComponent) podSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
+	psp := podsecuritypolicy.NewBasePolicy()
+	psp.GetObjectMeta().SetName(CSIDaemonSetName)
+	psp.Spec.Privileged = true
+	psp.Spec.AllowPrivilegeEscalation = ptr.BoolToPtr(true)
+	psp.Spec.Volumes = append(psp.Spec.Volumes, policyv1beta1.HostPath)
+	psp.Spec.RunAsUser.Rule = policyv1beta1.RunAsUserStrategyRunAsAny
+
+	return psp
+}
+
+func (c *csiComponent) clusterRole() *rbacv1.ClusterRole {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"users", "groups", "serviceaccounts"},
+			Verbs:     []string{"impersonate"},
+		},
+	}
+
+	// Allow access to the pod security policy in case this is enforced on the cluster
+	policyRules = append(policyRules, rbacv1.PolicyRule{
```

Comment:
Should we put this behind the `UsePSP` flag?

Reply:
left this out at first since I put the whole role behind the UsePSP  - but moved it under usepsp https://github.com/tigera/operator/pull/2226/commits/24eaef7a516b0d15ca1400586e78a85655fc2c79 in case we need to reuse the role